File: dojo_9/java/src/main/java/com/mercadolibre/dojos/Precio.java
package com.mercadolibre.dojos;

public class Precio {
    private Double monto;

    public Precio(double monto) {
        this.monto = monto;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj.getClass().equals(this.getClass())) {
            return this.monto.intValue() == ((Precio) obj).monto.intValue();
        } else {
            return false;
        }
    }

    public Precio plus(Precio precio) {
        this.monto += precio.monto;
        return this;
    }

    public Precio multiply(double factor) {
        this.monto *= factor;
        return this;
    }

    @Override
    public String toString() {
        return String.valueOf(monto.intValue());
    }
}

File: dojo_kotlin/app/src/main/java/com/nelsito/kotlindojo9/Precio.kt
package com.nelsito.kotlindojo9

/**
 * Created by npuglisi on 12/12/17.
 */
class Precio(var monto: Double) {
    fun plus(precio: Precio): Precio {
        this.monto = this.monto.plus(precio.monto)
        return this
    }
    fun multiply(factor: Int): Precio {
        this.monto = monto.times(factor)
        return this
    }

    override fun toString(): String {
        return monto.toString()
    }

    override fun equals(other: Any?): Boolean {
        if (other === this) return true
        if (other?.javaClass != javaClass) return false
        other as Precio
        return other.monto.toInt().equals(this.monto.toInt())
    }
}